codec-http/src/main/java/io/netty/handler/codec/http/HttpScheme.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2015 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.codec.http;
+
+import io.netty.util.AsciiString;
+
+/**
+ * Defines the common schemes used for the HTTP protocol as defined by
+ * <a href="https://tools.ietf.org/html/rfc7230">rfc7230</a>.
+ */
+public final class HttpScheme {
```

Comment:
It is used by this PR. It is easier to leave it in (less PR dependencies, and merge forward to master will be self contained), but I'll separate out the different components of this PR.